DEVELOPMENT.md

```diff
@@ -714,10 +714,78 @@ The message definitions in `pkg/webui/locales` can be used to provide translatio
 
 ## Testing
 
+### Unit Tests
+
+To run unit tests, observe the following mage targets:
+
 ```bash
 $ tools/bin/mage go:test js:test jsSDK:test
 ```
 
+### End-to-end Tests
+
+To run end-to-end tests, observe the following mage target:
+
+```bash
+$ ./mage endToEnd:run
+```
+
+We use ['cypress'](https://cypress.io) for running frontend-based end-to-end tests. The tests specifications are located at `/cypress/integration`.
+
+#### Interactive Mode
+
+It is possible to run the tests in interactive mode, which is a helpful tool during development. To run the tests in interactive mode, you need to run the following command to prepare the environment:
+
+```bash
+$ ./mage endToEnd:prepare
+```
+This will build assets and set up the testing database. Then, you can run the stack in testing configuration via
+
+```bash
+$ ./mage endToEnd:startTestStack
+```
+
+> Note: This command will suppress the output of executed The Things Stack for LoRaWAN)
+
+> Also note: make sure that you don't have an instance of the The Things Stack for LoRaWAN already running
+
+Now, you can run the test in interactive mode via
+
+```bash
+$ ./mage endToEnd:runOpen
+```
+
+#### Writing End-to-End Tests
+
+##### Cypress Documentation
+
+Cypress has an [excellent documentation](https://docs.cypress.io/guides/overview/why-cypress.html), it is a recommended read before starting to write tests.
+
+##### Guding Principle
+
+We follow the [guiding principle of Kent C. Dodds](https://twitter.com/kentcdodds/status/977018512689455106) ([@kentcdodds](https://twitter.com/kentcdodds)) for writing useful tests:
```

Comment:
I don't think this particular twitter post provides enough detail and I don't think we should make *a* twitter post our guiding principle in general, but it's a nice quote and maybe just put a reference below it?